tests/unit/test_vtk.py

```diff
@@ -1,9112 +1,409 @@
+            },
+        )
+
+    for intf, intf_data in mdg.interfaces(return_data=True):
+        pp.set_state(
+            intf_data,
+            {
+                "dummy_scalar": np.zeros(intf.num_cells),
+                "unique_dummy_scalar": np.zeros(intf.num_cells),
+            },
+        )
+
+    # Fetch separate subdomains
+    subdomains_1d = mdg.subdomains(dim=1)
+    subdomains_2d = mdg.subdomains(dim=2)
+    sd_2d = subdomains_2d[0]
+
+    # Export data
+    save = pp.Exporter(
+        mdg,
+        setup.file_name,
+        setup.folder,
+        export_constants_separately=False,
+    )
+    save.write_vtu(
+        [
+            (subdomains_1d, "dummy_scalar"),
+            "dummy_vector",
+            "unique_dummy_scalar",
+            (sd_2d, "cc", sd_2d.cell_centers),
+        ]
+    )
+
+    # Check that exported vtu files and reference files are the same
+    for appendix in ["1", "2", "mortar_1"]:
+        assert _compare_vtu_files(
+            f"{setup.folder}/{setup.file_name}_{appendix}.vtu",
+            f"{setup.folder_reference}/mdg_3_grid_{appendix}.vtu",
+        )
+
+
+def test_constant_data(setup):
+    # Test of the Exporter functionality to distinguish between constant and non-constant
+    # data during exporting.
+
+    # Define grid
+    g = pp.StructuredTriangleGrid([3] * 2, [1] * 2)
+    g.compute_geometry()
+
+    # Define data
+    dummy_scalar = np.ones(g.num_cells) * g.dim
+    dummy_vector = np.ones((3, g.num_cells)) * g.dim
+
+    # Export data
+    save = pp.Exporter(
+        g,
+        setup.file_name,
+        setup.folder,
+    )
+    # Add additional constant data (cell centers)
+    save.add_constant_data([(g, "cc", g.cell_centers)])
+    save.write_vtu([("dummy_scalar", dummy_scalar), ("dummy_vector", dummy_vector)])
+
+    # Check that exported vtu files and reference files are the same
+    for appendix in ["2", "constant_2"]:
+        assert _compare_vtu_files(
+            f"{setup.folder}/{setup.file_name}_{appendix}.vtu",
+            f"{setup.folder_reference}/constant_data_test_grid_{appendix}.vtu",
+        )
+
+
+def test_fractures_2d(setup):
+    # Test of the export functionality of FractureNetwork2d.
+
+    # Define network
+    p = np.array([[0, 2, 1, 2, 1], [0, 0, 0, 1, 2]])
+    e = np.array([[0, 2, 3], [1, 3, 4]])
+    domain = {"xmin": -2, "xmax": 3, "ymin": -2, "ymax": 3}
+    network_2d = pp.FractureNetwork2d(p, e, domain)
+
+    # Define data
+    dummy_scalar = np.ones(network_2d.num_frac())
+    dummy_vector = np.ones((3, network_2d.num_frac()))
+    data = {"dummy_scalar": dummy_scalar, "dummy_vector": dummy_vector}
+
+    # Make directory if not existent
+    if not os.path.exists(setup.folder):
+        os.makedirs(setup.folder)
+
+    # Export data
+    network_2d.to_file(
+        setup.folder + setup.file_name + ".vtu",
+        data=data,
+    )
+
+    # Check that exported vtu file and reference file are the same
```

Comment:
So, this test tacitly assumes that Gmsh returns the same grid every time? If so, I think we should do ourselves a favor and add a comment that if this test breaks, but not those above, one should rerun the test with an earlier version of gmsh and, if that works, the change is quite likely due to external factors. Or something like that.

Reply:
For all instances, it is updated. Please check.